runtime/gateway.go

```diff
@@ -274,55 +281,57 @@ func (gateway *Gateway) setupConfig(config *StaticConfig) {
 	}
 }
 
-func (gateway *Gateway) setupMetrics(config *StaticConfig) error {
+func (gateway *Gateway) setupMetrics(config *StaticConfig) (err error) {
 	metricsType := config.MustGetString("metrics.type")
+	service := config.MustGetString("metrics.tally.service")
 
-	var metricsBackend tally.CachedStatsReporter
 	if metricsType == "m3" {
 		if gateway.metricsBackend != nil {
 			panic("expected no metrics backend in gateway.")
 		}
 
-		env := config.MustGetString("env")
-
-		commonTags := map[string]string{"env": env}
-		m3Backend, err := m3.NewReporter(m3.Options{
+		// TODO: Why aren't common tags emitted?
+		// NewReporter adds 'env' and 'service' common tags; and no 'host' tag.
+		commonTags := map[string]string{
+			"env": config.MustGetString("env"),
+		}
+		opts := m3.Options{
 			HostPorts:          []string{config.MustGetString("metrics.m3.hostPort")},
-			Service:            config.MustGetString("metrics.tally.service"),
+			Service:            service,
```

Comment:
Since `commonTags` does not work we should just not pass `Service` and `CommonTags`